astropy/utils/decorators.py

```diff
@@ -467,15 +477,19 @@ def wrapper(*args, **kwargs):
                 # parameter was renamed to newkeyword.
                 if old_name[i] in kwargs:
                     value = kwargs.pop(old_name[i])
-                    # Display the deprecation warning only when it's only
+                    # Display the deprecation warning only when it's not
                     # pending.
                     if not pending[i]:
-                        warnings.warn(
-                            '"{0}" was deprecated in version {1} '
-                            'and will be removed in a future version. '
-                            'Use argument "{2}" instead.'
-                            ''.format(old_name[i], since[i], new_name[i]),
-                            warning_type, stacklevel=2)
+                        message = ('"{0}" was deprecated in version {1} '
+                                   'and will be removed in a future version. '
+                                   .format(old_name[i], since[i]))
+                        if new_name[i] is not None:
+                            message += ('Use argument "{}" instead.'
+                                        .format(new_name[i]))
+                        elif alternative:
```

Comment:
So, if we go with my suggestion, I'd swap the two branches of this `if` statement.

Reply:
Was this addressed already?

Reply:
No, and currently I don't plan to as per the discussion above https://github.com/astropy/astropy/pull/8324#discussion_r258758232. I happy to though in case there is a clear use case comes up above.